From 4ee67a5725a069b5d180cd5d8af1d51fbb90a097 Mon Sep 17 00:00:00 2001
From: Trent Mick <trentm@gmail.com>
Date: Fri, 17 Feb 2017 17:17:51 -0800
Subject: [PATCH] joyent/node-triton#176 triton `--act-as` option was broken by
 joyent/node-triton#156 Reviewed by: Julien Gilli <julien.gilli@joyent.com>

---
 lib/config.js | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/lib/config.js b/lib/config.js
index 7ea03dd..f1cae3f 100644
--- a/lib/config.js
+++ b/lib/config.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright 2016 Joyent, Inc.
+ * Copyright 2017 Joyent, Inc.
  */
 
 /*
@@ -61,7 +61,8 @@ var PROFILE_FIELDS = {
     account: true,
     keyId: true,
     insecure: true,
-    user: true
+    user: true,
+    actAsAccount: true
 };
 
 
@@ -232,6 +233,7 @@ function validateProfile(profile, profilePath) {
         assert.optionalString(profile.user,
             profile.name === 'env' ? 'TRITON_USER or SDC_USER'
                 : 'profile.user');
+        assert.optionalString(profile.actAsAccount, 'profile.actAsAccount');
     } catch (err) {
         var msg = format('invalid %sprofile%s: %s',
             profile.name ? '"' + profile.name + '" ' : '',
-- 
2.21.0

